### ISSUE TITLE
Ability to use ":" instead of "@" in named args

### ISSUE THREAD

**Issue by KoNekoD** [jackc/pgx#2177]
Add ability to use ":" instead of "@" in named args

Sql linter in golang like this

**Comment by KoNekoD** [comment]
Just add ":" to here https://github.com/jackc/pgx/blob/master/named_args.go#L110

**Comment by KoNekoD** [comment]
PR https://github.com/jackc/pgx/pull/2178

### Before(bad)

![image](https://github.com/user-attachments/assets/f507c3b1-5c98-4ff8-9534-8c71f42b38b5)


### After (good)

![image](https://github.com/user-attachments/assets/437ba977-c1cf-4475-97f2-813cca8b0bab)

**Comment by KoNekoD** [comment]
@jackc I don't think it will bring any significant problems, but only solve the “mad linter in goland” problem.

In fact, your implementation provokes goland to warnings, wouldn't it be better to let users choose whether to continue using '@' or to refuse explicit conversion (it will be even a plus because of the cost of conversion itself), but it will be code without warnings.

What do you think about this improvement?

**Comment by KoNekoD** [comment]
@jackc  I see your point, I'll make the edits according to your concerns.

**Comment by KoNekoD** [comment]
@jackc Done! Commit 2aa476d2f92d5599543e5db4e8f6aa14d3cb9369

**Comment by jackc** [comment]
The code looks more solid now.

But as I said above, I want to keep the simplicity of only having a single delimiter for named arguments.

I think this support for `:` should be pulled out into your own library or application.

**Comment by KoNekoD** [comment]
@jackc Are you sure that library users will be comfortable being forced to adjust to your understanding of good? There seem to be some people who don't like it when their IDE frowns on '@' substitutions. Maybe we should pay attention to the convenience for the end user of the library?

**Comment by Nerobrain** [comment]
@KoNekoD Are you sure that library users will be comfortable being forced to adjust to your understanding of good? There seem to be some people who know difference between `errors` and `linter faults`. Maybe we should pay attention to the workload for the autor of the library?

**Comment by devsalesexpress** [comment]
> `:`é complicado porque também é usado para conversão de tipo, por exemplo `select somecol::text`. Até onde eu sei, #2178 falharia quando `::`fosse encontrado.
> 
> Mas mesmo que o analisador fosse corrigido, eu preferiria que houvesse apenas uma maneira de denotar um argumento nomeado.
> 
> No entanto, `pgx.[Strict]NamedArgs`não requer nenhuma funcionalidade interna do pgx. Ele simplesmente implementa a `pgx.QueryRewriter`interface. Se você realmente precisa usar `:`, você pode copiar `NamedArgs`para o seu projeto e alterá-lo lá.

Just check if the successor (the next character) is also a colon ':' if it is then it is a parameter so do not replace the value.

**Comment by KoNekoD** [comment]
I hope my option will still be considered suitable for merging, ':' this is the accepted standard

**Comment by jackc** [comment]
Sorry, but I still don't think supporting two bind variable styles is a good idea, and I also don't want to be responsible for supporting the additional complexity in the SQL parser / rewriter.

But as I mentioned above, the is nothing in `pgx.[Strict]NamedArgs` requires access to pgx internals. It simply implements the `pgx.QueryRewriter` interface. I would suggest creating a project for this alternative `NamedArgs` implementation. I'll even link to it from the README.md if you want.

**Comment by KoNekoD** [comment]
@jackc Done; Sorry for long delay: https://github.com/jackc/pgx/issues/2377

Project: https://github.com/KoNekoD/pgx-colon-query-rewriter

**Comment by jackc** [comment]
Just added the link.
